### PR TITLE
SLICK-506 Multiple Devices in Page Boot

### PR DESCRIPTION
**Pull request by vkimcm**

### CODE REVIEW

File: svn/trunk/SlickEngagement_Plugin.php [hunk 1]
```diff
@@ -288,10 +288,33 @@ private function echoSlickBootJs($boot_data_obj) {
         $this->echoSlickstreamComment('END Page Boot Data', false);
     }
 
+    private function isMobile() {
```
Comment (ericperez) [inline svn/trunk/SlickEngagement_Plugin.php:291]: Is the result of running this function always going to match the result of running the equivalent "is mobile" logic on the server side (for generating the v2 page-boot-data)?

cc: @mstreinz  Tagging you to make sure you see this. If this check isn't exactly what the server-side is doing for "mobile detection" then this side and/or the server side need to be updated to make them match.

File: svn/trunk/SlickEngagement_Plugin.php [hunk 2]
```diff
@@ -288,10 +288,33 @@ private function echoSlickBootJs($boot_data_obj) {
...
+        return $boot_data_obj;
+    }
+
     private function echoClsData($boot_data_obj)
     {
-        $filmstrip_config = isset($boot_data_obj->filmstrip) ? $boot_data_obj->filmstrip : '';
-        $dcm_config = isset($boot_data_obj->inlineSearch) ? $boot_data_obj->inlineSearch : '';
+        $boot_data = $this->getBootDataForDevice($boot_data_obj);
+
+        $filmstrip_config = isset($boot_data->filmstrip) ? $boot_data->filmstrip : '';
```
Comment (ericperez) [inline svn/trunk/SlickEngagement_Plugin.php]: Shouldn't we be falling back on the config from `$boot_data_obj` in the case that for some reason `$boot_data->filmstrip` is not set? i.e.

`$filmstrip_config = isset($boot_data->filmstrip) ? $boot_data->filmstrip : $boot_data_obj->filmstrip;`

Same thing for `$dcm_config`.

File: svn/trunk/SlickEngagement_Plugin.php [hunk 3]
```diff
@@ -288,10 +288,33 @@ private function echoSlickBootJs($boot_data_obj) {
...
+            return $boot_data_obj_v2->desktop;
+        }
+        return $boot_data_obj;
+    }
+
     private function echoClsData($boot_data_obj)
     {
-        $filmstrip_config = isset($boot_data_obj->filmstrip) ? $boot_data_obj->filmstrip : '';
-        $dcm_config = isset($boot_data_obj->inlineSearch) ? $boot_data_obj->inlineSearch : '';
+        $boot_data = $this->getBootDataForDevice($boot_data_obj);
```
Comment (ericperez) [inline svn/trunk/SlickEngagement_Plugin.php]: Suggest renaming `$boot_data` to `$device_boot_data`.

File: svn/trunk/SlickEngagement_Plugin.php (hunk 1 above)
Comment (vkimcm) [reply svn/trunk/SlickEngagement_Plugin.php:291]: i copied isMobile function from server, are we changing it?

File: svn/trunk/SlickEngagement_Plugin.php (hunk 2 above)
Comment (vkimcm) [reply svn/trunk/SlickEngagement_Plugin.php]: i dont think it's necessary, boot_data will always be set, it is defaulted to our original config if v2 doesn't exist

File: svn/trunk/SlickEngagement_Plugin.php (hunk 3 above)
Comment (vkimcm) [reply svn/trunk/SlickEngagement_Plugin.php]: updated

File: svn/trunk/SlickEngagement_Plugin.php (hunk 1 above)
Comment (vkimcm) [reply svn/trunk/SlickEngagement_Plugin.php:291]: we are going to create separate ticket to update device detection later on and keep isMobile for now

**Review by mstreinz — APPROVED** [review]

**Pull request merged**